finality-aleph/src/party/mod.rs

```diff
@@ -470,7 +472,54 @@ where
         };
         loop {
             let last_finalized_number = self.client.info().finalized_number;
-            debug!(target: "afa", "Highest finalized: {:?} session {:?}", last_finalized_number, session_id);
+            if last_finalized_number > first_block {
+                let next_session_id = SessionId(session_id.0 + 1);
+                let next_session_authorities = match self
+                    .client
+                    .runtime_api()
+                    .next_session_authorities(&BlockId::Number(first_block))
```

Comment:
Can we create a new method of ConsensusParty that fetches the committee for a given **session**, updates `session_authorities` and outputs the result (this function should attempt using `next_session_authorities`  unless the session is `0`). This way we can make this code much nicer and also call this function at the start of `run_session` instead of in `run`.